server/remote_asset/fetch_server/fetch_server_test.go

```diff
@@ -184,10 +192,9 @@ func TestFetchBlobWithCache(t *testing.T) {
 			ctx, err := prefix.AttachUserPrefixToContext(ctx, te)
 			require.NoError(t, err)
 
-			err = te.GetCache().Set(ctx, digest.NewResourceName(&repb.Digest{
-				Hash:      tc.contentHash,
-				SizeBytes: contentSize,
-			}, "", resource.CacheType_CAS, tc.checksumFunc).ToProto(), []byte("content"))
+			contentDigest, err := digest.Compute(bytes.NewReader([]byte(content)), tc.checksumFunc)
+			require.NoError(t, err)
+			err = te.GetCache().Set(ctx, digest.NewResourceName(contentDigest, "", resource.CacheType_CAS, tc.checksumFunc).ToProto(), []byte(content))
```

Comment:
@sluongng this might be a pre-existing issue with the current test, but shouldn't we be storing to the cache with `tc.storageFunc` here rather than `tc.checksumFunc`? I thought the checksumFunc is only used when validating the content in the fetch server, and not used for caching purposes at all

Reply:
We are already storing the cache with `storageFunc`.

However, for cases where the `storageFunc != checksumFunc` (i.e. most of rules_js use SHA512 while Bazel runs on SHA256), subsequent FetchBlob requests get no cache hits.

This PR makes it so that we double write: one cache entry with `storageFunc` and one cache entry with `checksumFunc`. So subsequent requests would pull the entry from our cache using `checksumFunc` instead of fetching the blob from the internet.